acceptfunc.go

```diff
@@ -52,3 +57,43 @@ func defaultMsgAcceptFunc(dh Header) MsgAcceptAction {
 	}
 	return MsgAccept
 }
+
+type PacketAction int
+
+const (
+	Accept PacketAction = iota
+	Reject
+)
+
+type AcceptOption struct {
+	Last time.Time
+}
+
+type PacketAccepter interface {
+	Accept(*AcceptOption) PacketAction
+}
+
+// our default implementation
+
+type PacketAccept struct {
+	Max int // Max is the maximum number of goroutines to allow.
+}
+
+func (p PacketAccept) Accept(opt *AcceptOption) PacketAction {
+	numgo := runtime.NumGoroutine()
```

Comment:
If we want people to respond to the number of goroutines, wouldn't it make more sense to internally track the number of goroutines *we*-create (which sync.WaitGroup is doing internally). Otherwise this could get confused by goroutines spawned by other code in the binary (including the runtime itself).